nodejs_tests/src/lib.rs

```diff
@@ -78,16 +78,37 @@ fn call_with_42 (
     retain: unsafe extern "C" fn(data: *mut ::std::os::raw::c_void),
 ) -> u8
 {
-    retain(data);
-    ::std::thread::spawn({
-        let data = data as usize;
-        move || {
-            dbg!(cb(data as _, 42));
-            release(data as _);
-        }
-    });
-
-    let ret = dbg!(cb(data, 42));
+    thread_local! {
+        static CB
+            : ::core::cell::Cell<
+                Option<(
+                    *mut ::std::os::raw::c_void,
+                    unsafe extern "C" fn(data: *mut ::std::os::raw::c_void, x: i32) -> u8,
+                    unsafe extern "C" fn(data: *mut ::std::os::raw::c_void),
+                )>
+            >
+            = None.into()
+        ;
```

Comment:
Nevermind the "clarity" of this file, I have just used it to write potentially weird patterns that happen to cover / challenge most edge cases of the FFI.
In this instance, within this `call_with_42` callback, I am testing the three main situations where a `JsFunction` may be called:
  - same-thread, original FFI-call,
  - background thread,
  - same-thread, follow-up FFI call (hence this thread_local)